site/en/docs/extensions/reference/storage/index.md

```diff
@@ -128,17 +174,17 @@ to this setting are immediately saved to sync storage by the options page and th
 ```js
 // In-page cache of the user's options
 const options = {};
+const optionsForm = document.getElementById("optionsForm");
 
 // Initialize the form with the user's option settings
-chrome.storage.sync.get('options', (data) => {
-  Object.assign(options, data.options);
-  optionsForm.debug.checked = Boolean(options.debug);
-});
+const data = await chrome.storage.sync.get("options");
+Object.assign(options, data.options);
+optionsForm.debug.checked = Boolean(options.debug);
```

Comment:
I'm inclined to move these 3 lines below the event listener registration since registering a listener is synchronous, but fetching data from storage is async. We should prepare as much of the application as reasonably possible on the first turn of the event loop. Also, calling `optionsForm.debug.checked = Boolean(options.debug)` will not trigger a change event.

Reply:
Will do 👍